syntax = "proto3";

message EtfHoldings {
    string symbol = 1;
    string holding_name = 2;
    float holding_percent = 3;
}

message SectorWeights {
    string sector_name = 1;
    float sector_weight = 2;
}

message Etf {
    string yahoo_symbol = 1;
    string name = 2;
    string currency = 3;
    repeated EtfHoldings  holdings = 4;
    repeated SectorWeights weights = 5;
    int64 total_assets = 6;
    float expense_ratio = 7;
    float last_close = 8;
}

message GetEtfByName {
    string name = 1;
}

message EtfByTicker {
    string yahoo_symbol = 1;
}

service EtfService {
    rpc GetEtf (EtfByTicker) returns (Etf) {}
    rpc AddEtf (EtfByTicker) returns (Etf) {}
}
